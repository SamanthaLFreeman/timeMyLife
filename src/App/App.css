.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  height: 10vh;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: calc(8px + 2vmin);
  color: white;
}
