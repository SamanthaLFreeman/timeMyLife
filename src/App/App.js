import React, { Component } from 'react';
import ActivityForm from '../ActivityForm/ActivityForm';
import PastActivities from '../PastActivities/PastActivities';
import Timer from '../Timer/Timer';

class App extends Component {
  constructor() {
    super();
    this.state = {
      activities: [
        {
          category: 'Study',
          name: 'Test',
          mins: '05',
          secs: '30'
        }
      ],
      currentActivity: null,
      mins: null,
      secs: null
    }
  }

  setCurrentActivity = (currentActivity, mins, secs) => {
    this.setState({currentActivity, mins, secs});
  }

  removeCurrentActivity = () => {
    this.setState({
      currentActivity: null,
      mins: null,
      secs: null
    })
  }

  startTimer = () => {
    const { mins, secs } = this.state
    if(mins !== 0 && secs !== 0) {
    this.myInterval = setInterval(() => {
      if (secs > 0) {
        this.setState({
          secs: secs - 1
        })
      }
      if (secs === 0) {
        if (mins === 0) {
          clearInterval(this.myInterval)
        } else {
          this.setState({
            mins: mins - 1,
            secs: 59,

          })
        }
      }
    }, 1000)
  }
  }

  componentWillUnmount() {
    clearInterval(this.myInterval)
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1>Time My Life</h1>
        </header>
        <main className="main">
          {!this.state.currentActivity && <ActivityForm setCurrentActivity={this.setCurrentActivity} />}
          {this.state.currentActivity && <Timer removeCurrentActivity={this.removeCurrentActivity} currentActivity={this.state.currentActivity}
          mins={this.state.mins}
          secs={this.state.secs}
          startTimer={this.startTimer} />}
          <PastActivities activities={this.state.activities} />
        </main>
      </div>
    )
  }
}

export default App;
