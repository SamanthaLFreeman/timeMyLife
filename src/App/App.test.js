import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  it('should match the snapshot with the correct data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  })
})