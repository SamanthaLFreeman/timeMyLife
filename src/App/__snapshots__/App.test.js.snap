// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with the correct data passed through 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Time My Life
    </h1>
  </header>
  <main
    className="main"
  >
    <ActivityForm
      setCurrentActivity={[Function]}
    />
    <PastActivities
      activities={
        Array [
          Object {
            "category": "Study",
            "mins": "05",
            "name": "Test",
            "secs": "30",
          },
        ]
      }
    />
  </main>
</div>
`;
